### ISSUE TITLE
If I implement IObservable<T>, IObserver<T> in my own class, its a false negative on compatibility

### ISSUE THREAD

**Issue by sharpe5** [Microsoft/dotnet-apiport#373]
I am currently porting my library NetMQ.ReactiveExtensions to .NET Core:
https://github.com/NetMQ/NetMQ.ReactiveExtensions

However, it is currently giving a false negative on the compatility with .NET Core 1.0 and .NET 4.6.2.

I implemented IObservable<T>, IObserver<T> in my own class:

![iob](https://cloud.githubusercontent.com/assets/1290916/20076777/9f7b6aca-a531-11e6-902a-9a61735835f2.png)

Now, whenever I consume that class elsewhere, it says that this call is not compatible with anything - even .NET 4.6.2!

![sub](https://cloud.githubusercontent.com/assets/1290916/20076884/fd9d35d4-a531-11e6-9717-d391eff93609.png)

Here is the report. I can assure you, this library works well in .NET 4.6.2 :)

![portab](https://cloud.githubusercontent.com/assets/1290916/20077019/8f80446e-a532-11e6-9b50-1276bd38dceb.png)

**Comment by conniey** [comment]
@sharpe5 Thanks for the issue report! Would you mind passing me a small repro? According to the API Catalog, there is only one API in `IObservable<T>`, that is: `IObservable<T>.Subscribe(IObserver<T> observer);`.  So I am curious where this API definition comes from.

Cheers,
Connie

**Comment by sharpe5** [comment]
I implemented IObservable and  IObserver myself. I used RX for just the
interfaces. The implementation allows RX messages to be sent between
different processes using ZeroMQ as the transport/serialization layer.

See https://github.com/NetMQ/NetMQ.ReactiveExtensions.

I guess you could say that the  interface comes from Microsoft, but I wrote
the implementation.

I'm currently in the process of porting everything to run on .NET Core.

**Comment by sharpe5** [comment]
Hi,

FYI, last night, I upgraded from RX 2.2.5 to RX 3.1.0. I also broke the
build (temporarily).

You can check the code out and run ApiPort on it by ignoring any checkins
in the last 24 hours:
https://github.com/NetMQ/NetMQ.ReactiveExtensions

**Comment by conniey** [comment]
@sharpe5 Hi, I did the following:

1. `git clone https://github.com/NetMQ/NetMQ.ReactiveExtensions`
2. Build solution
3. Right-click -> "Analyze Assembly Portability" (my preferences were against .NET 4.6.2, .NET Core App 1.1, 1.2, .NET Standard 1.3, 1.6)
4. Viewed report

I cannot see that error you are seeing. :(

**Comment by conniey** [comment]
It may be helpful to note that our API Catalog only indexes APIs that the .NET Framework ships. So APIs in MEF and Reactive .NET are probably not part of the catalog.

**Comment by sharpe5** [comment]
Please pull a checkin that is more than 48 hours old. Last night, I
upgraded from RX 2.2.5 to RX 3.1.0. Can reply in more detail tonight.

**Comment by conniey** [comment]
@sharpe5 Hey, I pulled your repository 30 minutes ago and it says that the last checkin was "Tue Nov 8 21:17:12 2016 +0000".

How we determine whether to send an API to our service is based on this [`DotNetFrameworkFilter.cs`](https://github.com/Microsoft/dotnet-apiport/blob/a5988f7c0cd01ce31b83ea7bdee39b8abf4d4c38/src/Microsoft.Fx.Portability/Analyzer/DotNetFrameworkFilter.cs#L37-L62). It is probably the case that your assembly does not pass this filter that we send the API in and try to see if the API is in one of our known assemblies. if you want this to be ignored, you may want to change the filter to include APIs with a certain namespace.

**Comment by sharpe5** [comment]
Hi,

By the way, awesome video on Channel 9, you guys are doing a great job.
Some feedback: my whole aim is to make my NuGet package compatible with
.NET Core. Making the API calls compatible was only 50% of the job. The
other 50% was getting the .nuspec file compatible. It would be awesome if
you could auto-analyze any NuGet package definition files, and advise on
fixes to make them compatible. The thing that really tripped me up for a
long time was that I had to target "netstandard" (or "netcore") rather than
"net45". I eventually found this rather obscure page that documents this,
but only after I had found the solution almost by a lot of trial and error:
https://docs.nuget.org/ndocs/schema/target-frameworks

An automatic hint on how to fix this issue in any detected .NuSpec files
would save soooo much time for all the developers out there!

Do you want me to open an issue for this?

Best,
Shane

**Comment by sharpe5** [comment]
Hi,

Ok, I'll prepare a repository to reproduce the problem (am on the train to
work now). You will have it within 24 hours.

Best,
Shane

**Comment by sharpe5** [comment]
Closed, as I could not reproduce this issue.

I went back through all of the checkins, did a "git reset --hard <ID>", and all of them passed with no problems.

I think that either:
(a) The issue has been fixed server side, or;
(b) It was a rare corner case when compiling for .NET Standard, but still referencing RX 2.2.5. In this case, the failure was genuine. As RX references are now updated to RX 3.1.0, this should not be an issue moving forward.
